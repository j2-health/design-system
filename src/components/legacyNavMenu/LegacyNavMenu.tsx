import React from 'react'
import styles from './LegacyNavMenu.module.css'
import J2Logo from '../../svgs/j2-logo.svg?react'
import { UserProfile } from './UserProfile'

export interface NavItem {
  id: string
  label: string
  icon?: React.ReactNode
  isActive?: boolean
}

export interface NavSection {
  title: string
  items: NavItem[]
}

export interface LegacyNavMenuUser {
  firstName: string
  lastName?: string
  avatarUrl?: string
}

export interface LegacyNavMenuProps {
  sections: NavSection[]
  user: LegacyNavMenuUser
  onItemClick?: (itemId: string) => void
  logoUrl?: string
  onLogOut?: () => void
  onChangeClient?: () => void
  headerSlot?: React.ReactNode
  footerSlot?: React.ReactNode
}

export const LegacyNavMenu: React.FC<LegacyNavMenuProps> = ({
  sections,
  user,
  onItemClick,
  logoUrl = '/',
  onLogOut,
  onChangeClient,
  headerSlot,
  footerSlot,
}) => {
  const handleItemClick = (itemId: string) => {
    onItemClick?.(itemId)
  }

  return (
    <div className={styles.navContainer}>
      {/* Logo Section */}
      <div className={styles.logoSection}>
        <div className={styles.logoContainer}>
          <a href={logoUrl} className={styles.logoLink}>
            {/* @ts-expect-error - SVG component props */}
            <J2Logo className={styles.logo} />
          </a>
        </div>
      </div>

      {headerSlot && <div className="px-4">{headerSlot}</div>}

      {/* Navigation Sections */}
      <div className={styles.navContent}>
        {sections.map((section) => (
          <div key={section.title} className={styles.navSection}>
            {/* Section Header */}
            <div className={styles.sectionHeader}>
              <span className={styles.sectionTitle}>{section.title}</span>
            </div>

            {/* Section Items */}
            <div className={styles.sectionItems}>
              {section.items.map((item) => (
                <button
                  key={item.id}
                  className={`${styles.navItem} ${item.isActive ? styles.active : ''}`}
                  onClick={() => handleItemClick(item.id)}
                >
                  {item.icon && (
                    <div className={styles.navItemIcon}>{item.icon}</div>
                  )}
                  <span className={styles.navItemLabel}>{item.label}</span>
                </button>
              ))}
            </div>
          </div>
        ))}
      </div>

      {footerSlot && <div className="px-4">{footerSlot}</div>}

      {/* User Profile Section */}
      <UserProfile
        firstName={user.firstName}
        lastName={user.lastName}
        avatarUrl={user.avatarUrl}
        onLogOut={onLogOut}
        onChangeClient={onChangeClient}
      />
    </div>
  )
}
