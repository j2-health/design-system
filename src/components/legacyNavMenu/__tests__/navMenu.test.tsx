import { act, render } from '@testing-library/react'
import { LegacyNavMenu } from '../LegacyNavMenu'

describe('LegacyNavMenu', () => {
  it('should render correctly', async () => {
    const { container } = await act(async () => render(<LegacyNavMenu />))

    expect(container).toMatchSnapshot()
  })
})
