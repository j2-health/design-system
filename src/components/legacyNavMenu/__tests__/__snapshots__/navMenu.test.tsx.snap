// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegacyNavMenu should render correctly 1`] = `
<div>
  <div
    class="navContainer"
  >
    <div
      class="logoSection"
    >
      <div
        class="logoContainer"
      >
        <a
          class="logoLink"
          href="/"
        >
          <test-file-stub
            classname="logo"
          />
        </a>
      </div>
    </div>
    <div
      class="navContent"
    >
      <div
        class="navSection"
      >
        <div
          class="sectionHeader"
        >
          <span
            class="sectionTitle"
          >
            ANALYZE
          </span>
        </div>
        <div
          class="sectionItems"
        >
          <button
            class="navItem "
          >
            <div
              class="navItemIcon"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm71.87,53.27L136,114.14V40.37A88,88,0,0,1,199.87,77.27ZM120,40.37v83l-71.89,41.5A88,88,0,0,1,120,40.37ZM128,216a88,88,0,0,1-71.87-37.27L207.89,91.12A88,88,0,0,1,128,216Z"
                />
              </svg>
            </div>
            <span
              class="navItemLabel"
            >
              Scorecard
            </span>
          </button>
          <button
            class="navItem "
          >
            <div
              class="navItemIcon"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M213.66,82.34l-56-56A8,8,0,0,0,152,24H56A16,16,0,0,0,40,40V216a16,16,0,0,0,16,16H200a16,16,0,0,0,16-16V88A8,8,0,0,0,213.66,82.34ZM160,51.31,188.69,80H160ZM200,216H56V40h88V88a8,8,0,0,0,8,8h48V216Zm-32-80a8,8,0,0,1-8,8H96a8,8,0,0,1,0-16h64A8,8,0,0,1,168,136Zm0,32a8,8,0,0,1-8,8H96a8,8,0,0,1,0-16h64A8,8,0,0,1,168,168Z"
                />
              </svg>
            </div>
            <span
              class="navItemLabel"
            >
              Reports
            </span>
          </button>
        </div>
      </div>
      <div
        class="navSection"
      >
        <div
          class="sectionHeader"
        >
          <span
            class="sectionTitle"
          >
            MANAGE
          </span>
        </div>
        <div
          class="sectionItems"
        >
          <button
            class="navItem active"
          >
            <div
              class="navItemIcon"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M232,112H136V88h8a16,16,0,0,0,16-16V40a16,16,0,0,0-16-16H112A16,16,0,0,0,96,40V72a16,16,0,0,0,16,16h8v24H24a8,8,0,0,0,0,16H56v32H48a16,16,0,0,0-16,16v32a16,16,0,0,0,16,16H80a16,16,0,0,0,16-16V176a16,16,0,0,0-16-16H72V128H184v32h-8a16,16,0,0,0-16,16v32a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V176a16,16,0,0,0-16-16h-8V128h32a8,8,0,0,0,0-16ZM112,40h32V72H112ZM80,208H48V176H80Zm128,0H176V176h32Z"
                />
              </svg>
            </div>
            <span
              class="navItemLabel"
            >
              Networks
            </span>
          </button>
        </div>
      </div>
      <div
        class="navSection"
      >
        <div
          class="sectionHeader"
        >
          <span
            class="sectionTitle"
          >
            ACCOUNT
          </span>
        </div>
        <div
          class="sectionItems"
        >
          <button
            class="navItem "
          >
            <div
              class="navItemIcon"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M140,180a12,12,0,1,1-12-12A12,12,0,0,1,140,180ZM128,72c-22.06,0-40,16.15-40,36v4a8,8,0,0,0,16,0v-4c0-11,10.77-20,24-20s24,9,24,20-10.77,20-24,20a8,8,0,0,0-8,8v8a8,8,0,0,0,16,0v-.72c18.24-3.35,32-17.9,32-35.28C168,88.15,150.06,72,128,72Zm104,56A104,104,0,1,1,128,24,104.11,104.11,0,0,1,232,128Zm-16,0a88,88,0,1,0-88,88A88.1,88.1,0,0,0,216,128Z"
                />
              </svg>
            </div>
            <span
              class="navItemLabel"
            >
              User Guide
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="userSection"
    >
      <button
        class="userProfile"
      >
        <div
          class="userAvatar"
        >
          R
        </div>
        <span
          class="userName"
        >
          Rachel Foley
        </span>
      </button>
    </div>
  </div>
</div>
`;
